### PR TITLE
Disable embedded opencl platforms

### PR DESCRIPTION
**Pull request by jenshannoschwalm**
embedded platform might not be common around for machines wanting to run dt but there are a number of restrictions that make dt not working correctly. (or we would have to do a large number of tests)
(see https://www.khronos.org/registry/OpenCL/specs/3.0-unified/html/OpenCL_API.html#opencl-embedded-profile)

Also some more -d opencl corrections done.

### CODE REVIEW

**Comment by parafin** [comment]
Out of the interest: do you have an example of OpenCL implementation with embedded profile?

**Comment by jenshannoschwalm** [comment]
> Out of the interest: do you have an example of OpenCL implementation with embedded profile?

Not personally :-) Some snapdragons at least, see https://developer.qualcomm.com/software/adreno-gpu-sdk/gpu

File: src/common/opencl.c [hunk 1]
```diff
@@ -324,6 +323,7 @@ static int dt_opencl_device_init(dt_opencl_t *cl, const int dev, cl_device_id *d
                                            &(cl->dev[dev].max_mem_alloc), NULL);
   (cl->dlocl->symbols->dt_clGetDeviceInfo)(devid, CL_DEVICE_ENDIAN_LITTLE, sizeof(cl_bool), &little_endian, NULL);
 
+  cl->dev[dev].cltype = (unsigned int)type;
```
Comment (TurboGit) [inline src/common/opencl.c:326]: set here

File: src/common/opencl.h [hunk 2]
```diff
@@ -139,7 +139,10 @@ typedef struct dt_opencl_device_t
   int pinned_memory;
   // in OpenCL processing round width/height of global work groups to a multiple of this value.
   // reasonable values are powers of 2. this parameter can have high impact on OpenCL performance.
-  int clroundup;  
+  int clroundup;
+
+  // A bitfield that identifies the type of OpenCL device
+  unsigned int cltype;
```
Comment (TurboGit) [inline src/common/opencl.h:145]: declared/added here.

but never used. can you clarify?

**Review by TurboGit — COMMENTED** [review]
A question for something that looks suspicious to me.

File: src/common/opencl.h (hunk 2 above)
Comment (jenshannoschwalm) [reply src/common/opencl.h:145]: I want/need this for following pr requiering this info to disable "tuning cl device" if onboard graphics.
Either i remove this here and add to following pr or keep it as is.
As you want ...

Comment (TurboGit) [reply src/common/opencl.h:145]: No that's fine, just wanted to be sure it was not a missing commit for example. Thanks for the clarification.

**Review by TurboGit — APPROVED** [review]
Thanks!

**Pull request merged**